### PR TITLE
feat(ui): Add environment filtering on issue tag details page

### PR DESCRIPTION
**Pull request by scttcper**
This adds environment filtering on group tag details, previously environments were ignored. Closes https://getsentry.atlassian.net/browse/ISSUE-670

Adds an error message when the current environment selection has no results.

This gif shows:  
 1. navigating to a specific tag details
 2. changing environment to one that returns results
 3. changing environment to one with no results with new error message

Kapture 2020-05-11 at 12 05 44

### CODE REVIEW

File: src/sentry/api/endpoints/group_tagkey_values.py [hunk 1]
```diff
@@ -34,16 +35,10 @@ def get(self, request, group, key):
         """
         lookup_key = tagstore.prefix_reserved_key(key)
 
-        try:
-            environment_id = self._get_environment_id_from_request(
```
Comment (markstory) [inline src/sentry/api/endpoints/group_tagkey_values.py:38]: Are we able to delete the `_get_environment_id_from_request` method?

Comment (scttcper) [reply src/sentry/api/endpoints/group_tagkey_values.py:38]: nope, still used in ~14 areas

Comment (markstory) [reply src/sentry/api/endpoints/group_tagkey_values.py:38]: Darn.

Comment (markstory) [reply src/sentry/api/endpoints/group_tagkey_values.py:38]: It seems like we could make a separate pass to remove the `_get_environment_id_from_request` method. Both the endpoint method and the environment helper function are very similar and we don't really need to have both, but that is a task for another day :smile:

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx [hunk 2]
```diff
@@ -40,11 +42,16 @@ class GroupTagValues extends AsyncComponent<
...
+  getEndpoints(): [string, string, any?][] {
+    const {environments: environment} = this.props;
     const {groupId, tagKey} = this.props.params;
     return [
       ['tag', `/issues/${groupId}/tags/${tagKey}/`],
-      ['tagValueList', `/issues/${groupId}/tags/${tagKey}/values/`],
+      [
+        'tagValueList',
+        `/issues/${groupId}/tags/${tagKey}/values/`,
+        {query: {environment}},
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx:53]: Will this work with pagination?

Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx:53]: @billyvg It does seem to work with both environment and "cursor" parameters.

File: src/sentry/api/endpoints/group_tagkey_values.py [hunk 3]
```diff
@@ -34,16 +35,10 @@ def get(self, request, group, key):
...
-                request, group.project.organization_id
-            )
-        except Environment.DoesNotExist:
-            # if the environment doesn't exist then the tag can't possibly exist
-            raise ResourceDoesNotExist
+        environment_ids = [e.id for e in get_environments(request, group.project.organization)]
 
         try:
-            tagstore.get_tag_key(group.project_id, environment_id, lookup_key)
+            tagstore.get_tag_key(group.project_id, None, lookup_key)
```
Comment (wedamija) [inline src/sentry/api/endpoints/group_tagkey_values.py:41]: This is probably fine, but does change behaviour a little. Previously, we'd raise `ResourceDoesNotExist` here if the tag didn't exist within the environment, now we'll only raise if it doesn't exist at all. With your other changes I think this is ok though.

**Review by wedamija — APPROVED** [review]
Backend parts look good to me, probably worth getting a separate approval for the frontend pieces

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx [hunk 4]
```diff
@@ -59,6 +66,16 @@ class GroupTagValues extends AsyncComponent<
       property('count')
     ).reverse();
 
+    if (sortedTagValueList.length === 0 && this.props.environments?.length > 0) {
+      return (
+        <DetailedError
+          className="group-tags-details-error"
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx:72]: Is this class name used for anything?

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx [hunk 5]
```diff
@@ -59,6 +66,16 @@ class GroupTagValues extends AsyncComponent<
       property('count')
     ).reverse();
 
+    if (sortedTagValueList.length === 0 && this.props.environments?.length > 0) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx:69]: Can we destructure `environments` from `this.props` in the above block? https://github.com/getsentry/sentry/pull/18728/files#diff-95e6c286d2d062e433b73a138ffd114fR59-R62

Prop types suggests `environments` is required, do we need optional chaining?

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx (hunk 4 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx]: Removed!

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx (hunk 5 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx]: Thanks, good call out 👍

**Review by billyvg — APPROVED** [review]

**Pull request merged**